interfaces/prompting/errors.go

```diff
@@ -0,0 +1,198 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package prompting
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/snapcore/snapd/interfaces/prompting/errortypes"
+)
+
+var (
+	// NotFound errors when a prompt or rule is not found
+	ErrPromptNotFound = errors.New("cannot find prompt with the given ID for the given user")
+	ErrRuleNotFound   = errors.New("cannot find rule with the given ID")
+	ErrRuleNotAllowed = errors.New("user not allowed to request the rule with the given ID")
+
+	// Unsupported value errors, which are defined as functions below
+	ErrInvalidOutcome       = errInvalidOutcome
+	ErrInvalidLifespan      = errInvalidLifespan
+	ErrRuleLifespanSingle   = errRuleLifespanSingle
+	ErrInvalidInterface     = errInvalidInterface
+	ErrInvalidPermissions   = errInvalidPermissions
+	ErrPermissionsListEmpty = errPermissionsListEmpty
```

Comment:
why  both the exported var and the unexported implementation?

Reply:
The rationale was just to have the list of exported errors all in one place, regardless of whether the error is a function or a simple `errors.New()`. No functional reason, just ease of reading. I'm happy to just export the implementation if you'd prefer.